src/main/scala/cromwell/logging/WorkflowLogger.scala

```diff
@@ -0,0 +1,80 @@
+package cromwell.logging
+
+import akka.event.LoggingAdapter
+import cromwell.engine.WorkflowDescriptor
+import org.slf4j.Logger
+
+/**
+ * This sets up two loggers for a given workflow, specified by the
+ * WorkflowDescriptor.
+ *
+ * 1) the Akka event logger (optional).  The log messages sent to the Akka
+ *    logger will propagate to the root logger (see logback.xml)
+ *
+ * 2) A logger that writes to a file that only has log messages
+ *    for this particular workflow.  This makes it easier to debug
+ *    a particular workflow if the messages aren't interleaved with
+ *    other workflows.  This logger does NOT propagate to the root
+ *    logger
+ */
+case class WorkflowLogger[T](descriptor: WorkflowDescriptor, akkaLogger: Option[LoggingAdapter] = None, otherLoggers: Seq[Logger] = Seq.empty[Logger]) {
```

Comment:
For supplementary tag information like call name + index, we could have an implicit parameter so the tag doesn't need to be passed explicitly.  

In the caller:
```scala
  implicit val extraTag = SomeExtraLoggingTag(s"${call.name}:${call.index}")
```

Then log in the client without having to do `$tag` stuff every time.

Here:

```scala
sealed trait ExtraLoggingTag {
  def extraTag: String = ""
}
case object NoExtraLoggingTag extends ExtraLoggingTag
final case class SomeExtraLoggingTag(extra: String) extends ExtraLoggingTag {
  override def extraTag = extra
}

case class WorkflowLogger {
...
  private def makeTag(implicit extraLoggingTag: ExtraLoggingTag): String = ... // do stuff with WorkflowDescriptor + extraLoggingTag

  def warn(msg: String)(implicit extraLoggingTag: ExtraLoggingTag = NoExtraLoggingTag): Unit = {
    val fullMessage = makeTag + msg
    // log full mesage

}
```